### PR TITLE
Change initial_point default strategy from "prior" to "moment"

### LINKED ISSUE #5009: Change default initial_point method to "moment"
We set "prior" as default in #4983, but we actually want "moment". This makes the PR easier, because most distributions don't define `get_moment` yet, but as soon as that becomes available we should switch the default over.

### PR DESCRIPTION
**Pull request by ricardoV94**
This PR adds a check that we have moments for all distributions and changes the default initial point strategy from `"prior"` to `"moment"`

Also added a check in the default test moments that the logp of the provided moment is finite.

Closes #5009 
Related to #5078 

We also need to decide whether we default to a draw from the prior if a distribution does not have a `get_moment` implemented or if we just fail in that case (which is what is happening now). It would be trivial to allow this with a try/except block

### CODE REVIEW

**Comment by ricardoV94** [comment]
Getting some weird parallel sampling [test failures](https://github.com/pymc-devs/pymc/runs/4263439376?check_suite_focus=true) only on Windows. Perhaps related to https://github.com/pymc-devs/pymc/issues/4904?

**Comment by ricardoV94** [comment]
I am inclined to defaulting back to a prior sample if a distribution does not have a moment, perhaps with a UserWarning like

> Moment not defined for distribution X, defaulting to a draw from the prior. This can lead to difficulties during tuning. You can manually define an initval or implement a `get_moment` dispatched function for this distribution.

This would allow users to more gradually implement their own distributions. We still have a check added in this PR that all our standard distributions have moments implemented.

**Comment by michaelosthege** [comment]
I like the fallback plus warning idea. Maybe include a link to the relevant docs section on implementing custom distributions?

**Comment by ricardoV94** [comment]
We have very few moments missing, and it was consensus that this should no longer block the beta release. 

I have added the known missing moments to `test_all_distributions_have_moments`. This test will fail whenever those moments get implemented, making sure our coverage stays up to date.

**Comment by ricardoV94** [comment]
Still have those surprising failing tests on Windows alone...

**Comment by michaelosthege** [comment]
> Still have those surprising failing tests on Windows alone...

To me it looks like the same problem as #4904 - broadcasts are not correctly propagated through Cast operations.

**Comment by ricardoV94** [comment]
Windows issues are related to a mix of:

1. https://github.com/aesara-devs/aesara/issues/691,
2. https://github.com/aesara-devs/aesara/issues/692

**Comment by ricardoV94** [comment]
Tests are passing with the latest Aesara version

File: pymc/tests/test_distributions_moments.py [hunk 1]
```diff
@@ -62,13 +62,70 @@
...
+    # Assert that they do not have moments yet
+    assert missing_moments.intersection(not_implemented) == not_implemented
+
+    missing_moments -= not_implemented
+
+    if missing_moments:
+        raise NotImplementedError(
+            "The following distributions do not have a moment method:\n"
+            + "\n".join(sorted(map(str, list(missing_moments))))
+        )
```
Comment (michaelosthege) [inline pymc/tests/test_distributions_moments.py]: can we make the set operations a little easier to comprehend?

```suggestion
    unexpected_implemented = not_implemented - missing_moments
    unexpected_not_implemented = missing_moments - not_implemented
    
    if unexpected_implemented:
        raise Exception(
            f"Distributions {unexpected_implemented}" have a `get_moment` implemented."
            " That's good! But this test must be updated."
        )
        
    if unexpected_not_implemented:
        raise NotImplementedError(
            "Unexpected by this test, some distributions do not have a `get_moment` implementation:\n"
            + "\n".join(sorted(map(str, list(unexpected_not_implemented))))
        )
```

I didn't test this and my syntax probably doesn't pass the pre-commit. So please apply locally..

Comment (ricardoV94) [reply pymc/tests/test_distributions_moments.py]: Updated!

**Review by michaelosthege — APPROVED** [review]

**Pull request merged**